global_costmap:
  global_costmap:
    ros__parameters:
      update_frequency: 0.2
      publish_frequency: 0.2
      global_frame: map
      robot_base_frame: base_link
      use_sim_time: True
      # robot_radius: 0.32
      footprint: "[ [-0.05, 0.15], [0.40,0.15], [0.40,-0.15], [0.15,-0.15], [-0.05,-0.15] ]"
      # footprint: "[ [-0.05, 0.2], [0.5,0.2], [0.5,-0.2], [-0.05,-0.2] ]"
      footprint_padding: 0.05

      resolution: 0.05
      track_unknown_space: false # bke
      plugins: ["static_layer", "obstacle_layer", "inflation_layer"]
      obstacle_layer:
        plugin: "nav2_costmap_2d::ObstacleLayer"
        enabled: True
        observation_sources: scan
        scan:
          topic: /scan
          sensor_frame: laser_scanner_link
          max_obstacle_height: 2.0
          clearing: True
          marking: True
          data_type: "LaserScan"
          raytrace_max_range: 10.0
          raytrace_min_range: 0.2
          obstacle_max_range: 5.0
          obstacle_min_range: 0.2
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        map_subscribe_transient_local: True
      inflation_layer:
        plugin: "nav2_costmap_2d::InflationLayer"
        cost_scaling_factor: 2.0
        inflation_radius: 2.0
      always_send_full_costmap: True
  global_costmap_client:
    ros__parameters:
      use_sim_time: True
  global_costmap_rclcpp_node:
    ros__parameters:
      use_sim_time: True

map_server:
  ros__parameters:
    use_sim_time: True
    yaml_filename: "sim_orchard3.yaml"

guided_hybrid_planner_flow:
  ros__parameters:
    max_iterations: 10000000
    shot_distance: 0.5
    steer_angle_segment_size: 8
    wheel_base: 0.35
    cost_penalty: 0.0
    reverse_penalty: 3.0
    change_direction_penalty: 100.0
    steering_change_penalty: 2.0
    steering_penalty: 1.5
    max_steer_angle: 15.0
    publish_serch_tree: true
    show_log: false
    serch_radius: 4.0
    algo_type: 1
    vehicle_width: 0.2
    vehicle_length: 0.45
    vehicle_rear_axle_dist: 0.05

    smoother:
      max_iterations: 1000
      w_smooth: 0.1 # was 0.3
      w_data: 0.3 # was 0.2
      tolerance: 1e-10
      do_refinement: true
